### PR TITLE
pcre: adding with_stack_for_recursion option

### PR DESCRIPTION
**Pull request by whuji**
Specify library name and version:  **pcre/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Just adding a compilation flag support that I use into the recipe.

### CODE REVIEW

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/pcre/all/conanfile.py [hunk 1]
```diff
@@ -39,7 +40,8 @@ class PCREConan(ConanFile):
         "with_zlib": True,
         "with_jit": False,
         "with_utf": False,
-        "with_unicode_properties": False
+        "with_unicode_properties": False,
+        "with_stack_for_recursion": True
```
Comment (madebr) [inline recipes/pcre/all/conanfile.py:44]: I think the default should be `False`.
See https://github.com/LuaDist/pcre/blob/2b08a801c53f67b1d71fb83263fab819db78f886/CMakeLists.txt#L98
```suggestion
        "with_stack_for_recursion": False,
```

Comment (SpaceIm) [reply recipes/pcre/all/conanfile.py:44]: CMake option is NO_RECURSE while this recipe option is recurse actually, so it's fine.

Comment (whuji) [reply recipes/pcre/all/conanfile.py:44]: No it shouldn't.
With_stack_for_recursion at True, it puts "PCRE_NO_RECURSE" at OFF, and vice-versa.
PCRE_NO_RECURSE = **not** with_stack_for_recursion

When building with autotools, the default option is stack for recursion enabled (ie PCRE_NO_RECURSE at OFF).

Comment (madebr) [reply recipes/pcre/all/conanfile.py:44]: Haha, thanks!
You are both right!

**Review by madebr — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**